### PR TITLE
feat: implement license range intersection check

### PR DESCRIPTION
**Pull request by taleinat**
This adds range intersection checks, which are currently unimplemented.

### CODE REVIEW

File: test/ranges/intersects.test.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  const res1 = intersects(r1, r2);
+  const res2 = intersects(r2, r1);
+  if (res1 !== res2) {
+    throw new Error(`Inconsistent result: ${r1} vs. ${r2}`)
+  }
+  return res1;
+}
+
+test('intersects(range1, range2)', t => {
+  t.true(testIntersects('1.1', '>= 1.1'));
```
Comment (darmalovan) [inline test/ranges/intersects.test.js:17]: @taleinat are those tests based on other semver libs? just asking if we took inspiration from those.

File: lib/ranges.js [hunk 2]
```diff
@@ -75,3 +76,67 @@ function maxSatisfying(versions, range) {
 function minSatisfying(versions, range) {
   return _firstSatisfying(versions, range, compare);
 }
+
+function intersects(r1, r2) {
+  try {
+    let range1 = _expandTildes(_createRequirement(r1));
```
Comment (darmalovan) [inline lib/ranges.js]: `const` here ?

File: lib/ranges.js [hunk 3]
```diff
@@ -75,3 +76,67 @@ function maxSatisfying(versions, range) {
...
+    let range2 = _expandTildes(_createRequirement(r2));
```
Comment (darmalovan) [inline lib/ranges.js]: Same here

File: test/ranges/intersects.test.js (hunk 1 above)
Comment (taleinat) [reply test/ranges/intersects.test.js:17]: Yes, I took inspiration from tests in one of our other private semver-like libraries.

File: lib/ranges.js (hunk 2 above)
Comment (taleinat) [reply lib/ranges.js]: Certainly.

File: lib/ranges.js (hunk 3 above)
Comment (taleinat) [reply lib/ranges.js]: Done.

**Comment by taleinat** [comment]
@darmalovan, I've addressed your review comments, and this is ready for another look.

**Pull request merged**